vllm/v1/engine/core.py

```diff
@@ -134,21 +134,33 @@ def _initialize_kv_caches(
             self, vllm_config: VllmConfig) -> tuple[int, int, KVCacheConfig]:
         start = time.time()
 
-        # Get all kv cache needed by the model
-        kv_cache_specs = self.model_executor.get_kv_cache_specs()
-
-        # Profiles the peak memory usage of the model to determine how much
-        # memory can be allocated for kv cache.
-        available_gpu_memory = self.model_executor.determine_available_memory()
-
-        assert len(kv_cache_specs) == len(available_gpu_memory)
-        # Get the kv cache tensor size
-        kv_cache_configs = [
-            get_kv_cache_config(vllm_config, kv_cache_spec_one_worker,
-                                available_gpu_memory_one_worker)
-            for kv_cache_spec_one_worker, available_gpu_memory_one_worker in
-            zip(kv_cache_specs, available_gpu_memory)
-        ]
+        #TODO: CP start from here
+        if vllm_config.model_config.is_attention_free:
```

Comment:
I prefer to handle attention free model here https://github.com/vllm-project/vllm/blob/3eb4ad53f3e4ed0d3224c37efd08b4a1a42076fc/vllm/v1/core/kv_cache_utils.py#L927 . Would be great if it can be achieved by another branch in addtion to `is_kv_cache_type_uniform` and `is_kv_cache_page_size_uniform ` and pls tell me if further modfications are needed.